java/ruleXtract/src/it/java/uk/ac/cam/eng/extraction/RuleExtractorTest.java

```diff
@@ -0,0 +1,138 @@
+/*******************************************************************************
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use these files except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2014 - Juan Pino, Aurelien Waite, William Byrne
+ *******************************************************************************/
+
+package uk.ac.cam.eng.extraction;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.MapWritable;
+import org.apache.hadoop.io.SequenceFile;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import uk.ac.cam.eng.extraction.hadoop.datatypes.TextArrayWritable;
+import uk.ac.cam.eng.extraction.hadoop.features.phrase.Source2TargetJob;
+import uk.ac.cam.eng.extraction.hadoop.features.phrase.Target2SourceJob;
+import uk.ac.cam.eng.util.Pair;
+
+/**
+ * @author aaw35
+ *
```

Comment:
General request. Please add documentation to the code. It is not necessary to add per method, just one per class or at least per file, describing in some detail what is done.